source/sdk/lang/Array.h

```diff
@@ -12,20 +11,13 @@
 
 #include <stdint.h>
 
-
 #define _lang_array__Array_new(type, size) ((_lang_array__Array) { size, array_malloc((size) * sizeof(type)) });
 
 #if defined(safe)
 #define _lang_array__Array_get(array, index, type) ( \
-    (index < 0 || index >= array.length) ? \
-    lang_Exception__Exception_throw((lang_Exception__Exception *) lang_Exception__OutOfBoundsException_new_noOrigin(index, array.length)), \
-    *((type*) NULL) : \
-    ((type*) array.data)[index])
+    ((type*) array.data)[(index < 0 || index >= array.length) ? lang_Exception__Exception_throw((lang_Exception__Exception *) lang_Exception__OutOfBoundsException_new_noOrigin(index, array.length)), -1 : index])
```

Comment:
I removed the conditional expression and replaced it with simple array access, but the index calculation can throw exception now.
`-1` is there to indicate invalid index, array at `-1` is never accessed because exception is thrown. It uses the comma operator to first throw exception and then return correct type for array index.
```C
x = a, b;
```
x has the value and type of b but a is evaluated first